sources/packages/api/src/services/designation-agreement/designation-agreement.service.ts

```diff
@@ -96,4 +96,49 @@ export class DesignationAgreementService extends RecordDataModelService<Designat
       .andWhere("location.institution.id = :institutionId", { institutionId })
       .getOne();
   }
+
+  /**
+   * Get the summary of all designation agreements for the institution.
+   * @param institutionId institution to retrieve the designations.
+   * @returns designations under the institution.
+   */
+  async getInstitutionDesignationsById(
+    institutionId: number,
+  ): Promise<DesignationAgreement[]> {
+    return this.repo
+      .createQueryBuilder("designation")
+      .select([
+        "designation.id",
+        "designation.designationStatus",
+        "designation.submittedDate",
+        "designation.startDate",
+        "designation.endDate",
+      ])
+      .where("designation.institution.id = :institutionId", { institutionId })
+      .orderBy(
+        `CASE designation.designationStatus
+            WHEN '${DesignationAgreementStatus.Pending}' THEN 1
+            WHEN '${DesignationAgreementStatus.Approved}' THEN 2
+            WHEN '${DesignationAgreementStatus.Declined}' THEN 3
+            ELSE 4
+          END`,
+      )
+      .addOrderBy("designation.submittedDate")
```

Comment:
Dont we need to add the submittedDate before the DesignationAgreementStatus? It sorts the designation list by the submittedDate first and then by the DesignationAgreementStatus. So in a way the latest list of pending designation agreement will be shown and not a mix of DesignationAgreementStatus in submitted date order.

Reply:
The requirement is to have the list ordered by Status and for the group of status order by the submission date.
I did not follow the comment entirely but the idea is to show all Pending (only one always), then all the Approved ones and last all the Declined ones. Am I missing something here?

Reply:
Thanks for the demo looks good, please share the sql if you can

Reply:
Sharing the SQL as we talked. Just added the "DESC" in the end that was missing.
![image](https://user-images.githubusercontent.com/61259237/153313244-b41273c4-1c92-49c1-ad98-4cec4815b529.png)